tools/testbench/topology.c

```diff
@@ -137,7 +137,7 @@ int load_buffer(void *dev, int comp_id, int pipeline_id,
 		struct snd_soc_tplg_dapm_widget *widget)
 {
 	struct sof *sof = (struct sof *)dev;
-	struct sof_ipc_buffer buffer;
+	struct sof_ipc_buffer buffer = {0};
```

Comment:
Isn't the job of `tplg_load_buffer()` to initialize `buffer`? What error does this silence?

Reply:
testbench does not use all the data members as some are specific to HW config. This was only missing from the two "IPCs" in this PR.

Reply:
I don't understand sorry, I don't have the full picture. What was the error reported and in which function?

Reply:
Use of uninitialized data in FW buffer logic due to testbench not initializing it. Not an issue with kernel as it clears. Testbench bug.

Reply:
I understood that the error was testbench specific, my concern is that this change hides any "real" initialization bug in `tplg_load_buffer` in the future.

Reply:
Got you now, yep the userspace tplg parser is not setting any variables if it cant find tokens for them. I will fix this as another PR as it means removing more testbench code here and also testing other tools that used the shared tplg parser.